File: LeetCode/0771_Jewels_and_Stones/solution.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function(jewels, stones) {
    let jew={}
    let stone={}
    for(let i=0;i<jewels.length; i++){
        jew[jewels[i]]= (jew[jewels[i]] || 0)+1
    }
    for(let i=0;i<stones.length; i++){
        stone[stones[i]]= (stone[stones[i]] || 0)+1
    }
    let count=0;
    for(let char in jew){
        if(stone[char] == undefined) continue;
        else  count+=stone[char]
    }
    return count;
};

File: LeetCode/0771_Jewels_and_Stones/solution.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        map<int,int> jew = {};
        map<int,int> stone = {};
        for (int i = 0; i < jewels.size(); i++) {
            jew[jewels[i]] = jew[jewels[i]] + 1;
        }
        for (int i = 0; i < stones.length(); i++) {
            stone[stones[i]] = stone[stones[i]]+ 1;
        }
        int count = 0;
        for (const auto &[key,val] : jew) {
            if (stone[key] == 0)
                continue;
            else
                count += stone[key];
        }
        return count;
    }
};